ModuleThreeFirstTaskConsole/FileSystemVisitor.cs

```diff
@@ -0,0 +1,196 @@
+﻿// <copyright file="FileSystemVisitor.cs" company="PlaceholderCompany">
+// Copyright (c) PlaceholderCompany. All rights reserved.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+
+namespace ModuleThreeFirstTaskConsole
+{
+    /// <summary>
+    /// Can read filesystem and show documents and packages in linear form.
+    /// </summary>
+    public class FileSystemVisitor
+    {
+        private const string STANDARDPATH = @"C:\";
+
+        private readonly IFileSystem _fileSystem;
+        private readonly Func<IFileSystemInfo, bool> _filter;
+        private readonly string _initialPath;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FileSystemVisitor"/> class.
+        /// </summary>
+        /// <param name="fileSystem">Files and directories.</param>
+        /// <param name="filter">Filter of files and directories.</param>
+        /// <param name="initialPath">Place in filesystem where to start searching.</param>
+        public FileSystemVisitor(
+            IFileSystem fileSystem,
+            Func<IFileSystemInfo, bool> filter,
+            string initialPath)
```

Comment:
This greatly restricts the use of the class. Why couldn't we add a parameter to the Search method?

Reply:
Fixed in a883af4